server_test.go

```diff
@@ -332,10 +366,13 @@ func TestHTTPUpgrader(t *testing.T) {
 				if test.badSecKey {
 					nonce = nonce[:nonceSize-1]
 				}
-				test.req.Header.Set(headerSecKey, string(nonce))
+				if test.secKeyHeader == "" {
+					test.secKeyHeader = headerSecKey
+				}
+				test.req.Header[test.secKeyHeader] = []string{string(nonce)}
 			}
 			if test.err == nil {
-				test.res.Header.Set(headerSecAccept, string(makeAccept(test.nonce)))
+				test.res.Header[headerSecAccept] = []string{string(makeAccept(test.nonce))}
```

Comment:
Why these changes are needed?

Reply:
The `(http.Header) Set()` method calls `(textproto.MIMEHeader) Set()` which [canonicalizes the header key](https://golang.org/src/net/textproto/header.go?s=705:732) before setting the header. This causes tests to fail because the expected response (forced to lowercase-s headers) does not match the actual response (which now contains uppercase-S headers). We can bypass canonicalization by setting the key directly in the map that the textproto.MIMEHeader is based on.